module/os/freebsd/zfs/zfs_vfsops.c

```diff
@@ -1806,14 +1806,10 @@ zfs_fhtovp(vfs_t *vfsp, fid_t *fidp, int flags, vnode_t **vpp)
 	if (zfsvfs->z_parent == zfsvfs && fidp->fid_len == LONG_FID_LEN) {
 		zfid_long_t	*zlfid = (zfid_long_t *)fidp;
 		uint64_t	objsetid = 0;
-		uint64_t	setgen = 0;
 
 		for (i = 0; i < sizeof (zlfid->zf_setid); i++)
 			objsetid |= ((uint64_t)zlfid->zf_setid[i]) << (8 * i);
 
-		for (i = 0; i < sizeof (zlfid->zf_setgen); i++)
-			setgen |= ((uint64_t)zlfid->zf_setgen[i]) << (8 * i);
```

Comment:
There's also a good chance this is bug, the Linux counterpart to this function `zfs_vget` does make use of `setgen` for an error case.  This would also be a good candidate to move to it's own PR so we can get the FreeBSD folks to take a closer look.

Reply:
#12905